src/test/java/com/connexta/search/index/IndexTests.java

```diff
@@ -91,6 +92,33 @@ void testMissingFile() throws Exception {
         .andExpect(status().isBadRequest());
   }
 
+  @ParameterizedTest(name = "400 Bad Request if ID is {0}")
+  @NullSource
+  @ValueSource(
+      strings = {"   ", "1234567890123456789012345678901234", "+0067360b70e4acfab561fe593ad3f7a"})
```

Comment:
An advantage of making software components is that the test logic can focus on just the component. Components can be unit tested instead of having to test behavior through web services. It looks like most of what is exercised in this test is the plumbing-- the parts of Tomcat and Spring that handle HTTP requests and dispatch them to controllers. We generally assume the plumbing works and only test it in a few high-level tests. That leaves developers free to write more unit tests. The unit tests are simpler and faster to run as well as to write.

Reply:
I don't see what other "high-level tests" are verifying this if not this one. I also don't think we can make the assumption that "the plumbing works" because we had to build in a custom ExceptionHandler to handle this case, which wouldn't have been caught without a test like this.